File: src/main/java/com/procurement/evaluation/repository/AwardRepository.java
package com.procurement.evaluation.repository;

import com.procurement.evaluation.model.entity.AwardEntity;
import java.util.List;
import java.util.UUID;
import org.springframework.data.cassandra.repository.CassandraRepository;
import org.springframework.data.cassandra.repository.Query;
import org.springframework.stereotype.Repository;

@Repository
public interface AwardRepository extends CassandraRepository<AwardEntity, String> {

    @Query(value = "select * from evaluation_award where cp_id=?0 AND stage=?1 AND token_entity=?2 LIMIT 1")
    AwardEntity getByCpIdAndStageAndToken(String ocId, String stage, UUID token);

    @Query(value = "select * from evaluation_award where cp_id=?0 AND stage=?1")
    List<AwardEntity> getAllByCpidAndStage(String cpId, String stage);
}

File: src/main/kotlin/com/procurement/evaluation/application/repository/AwardRepository.kt
package com.procurement.evaluation.application.repository

import com.procurement.evaluation.domain.functional.Result
import com.procurement.evaluation.domain.model.Cpid
import com.procurement.evaluation.domain.model.award.AwardId
import com.procurement.evaluation.domain.model.enums.Stage
import com.procurement.evaluation.infrastructure.fail.Fail
import com.procurement.evaluation.model.entity.AwardEntity
import java.util.*

interface AwardRepository {
    fun findBy(cpid: String): List<AwardEntity>

    fun findBy(cpid: String, stage: String): List<AwardEntity>

    fun findBy(cpid: String, stage: String, token: UUID): AwardEntity?

    fun saveNew(cpid: String, award: AwardEntity)

    fun saveNew(cpid: String, awards: List<AwardEntity>)

    fun update(cpid: String, updatedAward: AwardEntity)

    fun update(cpid: String, updatedAwards: Collection<AwardEntity>)

    fun tryFindBy(cpid: Cpid, stage: Stage): Result<List<AwardEntity>, Fail.Incident>

    fun tryFindBy(cpid: Cpid, stage: Stage, awardId: AwardId): Result<AwardEntity?, Fail>

    fun trySave(cpid: Cpid, awards: List<AwardEntity>): Result<Unit, Fail.Incident>

    fun tryUpdate(cpid: Cpid, updatedAward: AwardEntity): Result<Boolean, Fail.Incident>
}
